src/main/java/org/ice4j/ice/Component.java

```diff
@@ -1207,17 +1212,53 @@ public Logger getLogger()
     public void send(byte[] buffer, int offset, int length)
             throws IOException
     {
-        CandidatePair pair = getSelectedPair();
+        CandidatePair pair = findPair(lastReceivedFrom);
```

Comment:
The naming of this code is confusing in the case when send-to-last-received-from-address isn't set.